vivisect/analysis/ppc/memorymaps.py

```diff
@@ -0,0 +1,67 @@
+# This analysis module initializes the initial PPC MMU entries which are
+# required for some of the embedded/bare-metal analysis in other analysis
+# modules
+#
+# For reference, this module uses information from the following reference
+# manuals:
+#   - "MPC5674F Microcontroller Reference Manual, rev 7" from NXP
+#   - "MPC5646C Microcontroller Reference Manual, rev. 5" from NXP
+#   - "MPC5777C Microcontroller Reference Manual, rev. 8" from NXP
+#
+import logging
+logger = logging.getLogger(__name__)
+
+import vivisect.exc as viv_exc
+import vivisect.const as viv_const
+
+def analyze(vw):
+    # For bare metal/embedded targets SPR reads and writes are important to
+    # track.
+    try:
+        vaset = vw.getVaSet("ppc_spr_reads")
+    except viv_exc.InvalidVaSet:
+        vw.addVaSet("ppc_spr_reads", (("funcva", viv_const.VASET_ADDRESS), ("reads", viv_const.VASET_COMPLEX)))
+
+    try:
+        vaset = vw.getVaSet("ppc_spr_writes")
+    except viv_exc.InvalidVaSet:
+        vw.addVaSet("ppc_spr_writes", (("funcva", viv_const.VASET_ADDRESS), ("writes", viv_const.VASET_COMPLEX)))
+
+    # Add in the initial MMU entries manually from the vivisect configuration
+    # options, or initialize a probably accurate enough default memory map
+    mmu_entries = vw.config.viv.arch.ppc.mmu
```

Comment:
I have confirmed that it can be set in a vivisect config, however I am unsure if there is a method that can be used to  pass complex structures on the command line that isn't horrible, or if there is a better way to configure this. I did find it necessary to be able to manually define the VLE/notVLE pages when binaries for different PowerPC processors because of how PowerPC determines what is and isn't a VLE memory page.

The True/False is used to indicate if a page is VLE or not, and it makes it easier to do automatic detection (if the pages were not specified on the command line and instead the PowerPC defaults are used).